frontend/src/pages/Graphing/components/Graph.tsx

```diff
@@ -1185,6 +1186,28 @@ const Graph = ({
 		false,
 	)
 
+	// Use a smaller bucketByWindow if the selected one is greater than the time range
+	if (
+		moment(startDate).add(bucketByWindow, 'second').isSameOrAfter(endDate)
+	) {
+		let lastPreset = DEFAULT_TIME_PRESETS[0]
+		for (const preset of DEFAULT_TIME_PRESETS) {
+			if (
+				moment(startDate)
+					.add(preset.quantity, preset.unit)
+					.isBefore(endDate)
+			) {
+				lastPreset = preset
+			} else {
+				break
+			}
+		}
+
+		bucketByWindow = moment
+			.duration(lastPreset.quantity, lastPreset.unit)
+			.asSeconds()
+	}
```

Comment:
https://www.loom.com/share/7b3be8b848b9411e891ffd7f987402a6

Reply:
able to repro this in preview as well. I think this is just the max buckets limitation - when you change bucket by to 1 second, it only loads the last 4 minutes of data (240 max buckets), and it looks like this metric is only reported every minute causing there to be only 3/4 data points in the graph

Reply:
oh, i misunderstood the PR (it's for the opposite case of the bucketing being too large for the time range, not for the case of the bucketing being too small to display all data)

Reply:
🚢